### PR TITLE
#3949 Pipeline page element property view (render with settings)

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3949 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    position: absolute;
+    right: 0px;
+    z-index: 1000;
+  `,
+);
+
+export const StyleScrollDiv = styled.div(
+  () => css`
+    height: 100%;
+    overflow-y: scroll;
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js]: 我們應該讓 Scroller 自行判斷是否要出現比較好 :)

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyView.js [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+        <StyleDisplay>
+          <Typography>{displayName}</Typography>
+          <Typography>{`Status is ${_.capitalize(cellStatus)}`}</Typography>
+        </StyleDisplay>
+        <IconButton className="close-button" onClick={handleClose}>
+          <CloseIcon />
+        </IconButton>
+      </StyleTitle>
+      <StyleScrollDiv>
+        <ExpansionPanel>
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyView.js]: 建議將 ExpansionPanel 的 border-radius 設為 0

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyView.js [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+          return <StorageIcon color="action" />;
+        }
+      default:
+        break;
+    }
+  };
+
+  return (
+    <StylePaper variant="outlined" square>
+      <StyleTitle>
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyView.js]: 我們可以讓 Scroller 的範圍也包含 Title 區塊嗎？

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    display: flex;
+    align-items: center;
+    height: 65px;
+    border-bottom: 1px solid ${theme.palette.divider};
+  `,
+);
+
+export const StyleIcon = styled.div(
+  ({ theme }) => css`
+    width: 25px;
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js]: 如果可以的話，應該儘量使用 8 的倍數

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    height: 25px;
+    border-radius: 100%;
+    background-color: #757575;
+    float: left;
+    margin: 0px 15px;
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js]: The same

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  () => css`
+    width: 60%;
+  `,
+);
+
+export const StyleSharedIconDiv = styled.div(
+  () => css`
+    position: absolute;
+    left: 17px;
+    top: 24px;
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js]: The same

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js (hunk 1 above)
Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js]: No problem. I'm thinking about using the third-party scrollbar which we already used in navigator for this.

**Comment by konekoya** [comment]
@eechih I'm taking over this PR since @wu87988622 is now helping me tackling #3813 :)

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyView.js (hunk 2 above)
Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyView.js]: Good suggestion :)

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyView.js (hunk 3 above)
Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyView.js]: Will do

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js (hunk 4 above)
Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelinePropertyView/PipelinePropertyViewStyles.js]: No problem

**Comment by eechih** [comment]
@konekoya 👍 整體看起來非常好～

以下有幾點小問題：
1. Topic 是否能顯示 Nodes 資訊？ 跟 Connector 有一致的行為
2. Properties View 的寬度，是否能讓使用者自由調整？
3. 數字是否能加上 comma？ 讓使用者容易閱讀，例如：1832138800 -> 1,832,138,800

**Comment by konekoya** [comment]
> Topic 是否能顯示 Nodes 資訊？ 跟 Connector 有一致的行為

I've asked @saivirtue about this. The node info are not available under topic API. 

> Properties View 的寬度，是否能讓使用者自由調整？
> 數字是否能加上 comma？ 讓使用者容易閱讀，例如：1832138800 -> 1,832,138,800
Good suggestion. Both are added :)

**Comment by konekoya** [comment]
> Topic 是否能顯示 Nodes 資訊？ 跟 Connector 有一致的行為

Okay, we have discussed this in the meeting. And since there are more info should be displayed for both topics and connectors. We will need to rethink about the UI and how should we display these info.

@eechih let me know if there's anything that should be addressed in this PR :)

File: ohara-manager/client/src/components/common/Definitions/Table.js [hunk 7]
```diff
@@ -110,7 +111,7 @@ const Table = props => {
                 setState(prevState => {
                   const data = [...prevState.data];
                   data.push(newData);
-                  onChange(data);
+                  onChange(data.map(row => omit(row, ['tableData'])));
```
Comment (eechih) [inline ohara-manager/client/src/components/common/Definitions/Table.js]: 建議使用 whitelist 會比較安全

Comment (konekoya) [reply ohara-manager/client/src/components/common/Definitions/Table.js]: Oops, this should be committed. @wu87988622 is currently working on this and should be commit in another PR. 😅

Comment (konekoya) [reply ohara-manager/client/src/components/common/Definitions/Table.js]: The code has been removed

**Comment by wu87988622** [comment]
@konekoya This pr is finish ?

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
> @konekoya This pr is finish ?

Not yet. Seems end-to-end is failing. I will have a look and fix it then i think it's ready to merge

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
@wu87988622 You could backport this to 0.9 now :)

**Pull request merged**